stdlib/tests/collections/smt.rs

```diff
@@ -269,61 +337,37 @@ fn assert_insert(
         new_root[2].as_int(),
         new_root[1].as_int(),
         new_root[0].as_int(),
-    ];
+    ]);
     let (store, adv_map) = build_advice_inputs(init_smt);
-    build_test!(source, &initial_stack, &[], store, adv_map).expect_stack(&expected_output);
+    let process = build_test!(source, &initial_stack, &[], store, adv_map.clone())
+        .execute_process()
+        .unwrap();
+
+    // check the returned values
+    let stack = stack_to_ints(&process.stack.trace_state());
+    assert_eq!(stack, expected_output);
+
+    // remove old values from the advice map
+    let mut new_adv_map = process.advice_provider.map().clone();
+    for (key, value) in adv_map.iter() {
+        let old_value = new_adv_map.remove(key).unwrap();
+        assert_eq!(value, &old_value);
+    }
+
+    // make sure the remaining values in the advice map are the same as expected new entries
+    assert_eq!(new_adv_map.len(), new_map_entries.len());
+    for (key, val) in new_map_entries {
+        let old_val = new_adv_map.get(key).unwrap();
```

Comment:
minor nit: I wonder whether or not `old_val` is appropriate as a name for this variable

Reply:
Yeah - I've replaced it with `init_value`.